src/Surfnet/Stepup/Identity/Identity.php

```diff
@@ -495,6 +502,53 @@ public function vetSecondFactor(
         );
     }
 
+    /**
+     * Self vetting, is when the user uses it's own token to vet another.
+     *
+     * Here the new token should have a lower or equal LoA to that of the one in posession of the user
+     */
+    public function selfVetSecondFactor(
+        Loa $authoringSecondFactorLoa,
```

Comment:
Could you elaborate why the id of the authoring second factor isn't used to query the Collection in $this->vettedSecondFactors for the required LOA?

Reply:
Sadly we have no knowledge of the authoring second factor other then the LoA it had. Gateway does not store the token ID used in it's assertion back to the SP (SelfService).